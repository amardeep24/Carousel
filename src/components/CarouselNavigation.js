import React from 'react'
import PropTypes from 'prop-types';
import NavigateBeforeIcon from '@material-ui/icons/NavigateBefore';
import NavigateNextIcon from '@material-ui/icons/NavigateNext';
import {
    NavigationButton,
    ButtonContainer
} from './Styled';

function CarouselNavigation(props) {
    const icon = props.direction === "left" ? <NavigateBeforeIcon /> : <NavigateNextIcon />;
    return (
        <ButtonContainer onClick={props.click}>
            <NavigationButton>{icon}</NavigationButton>
        </ButtonContainer>
    );
}

CarouselNavigation.propTypes = {
    direction: PropTypes.string
}

export default CarouselNavigation

