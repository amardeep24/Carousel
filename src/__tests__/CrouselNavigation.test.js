import React from 'react';
import { shallow, configure } from 'enzyme';
import CarouselNavigation from '../components/CarouselNavigation';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('Test 1::The CarouselNavigation component should match the snapshot', () => {
    let component;

    beforeEach(() => {
        component = shallow(<CarouselNavigation direction={"left"} />);
    });

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('Test 2::The CarouselNavigation component should render children properly', () => {
    let component;

    beforeEach(() => {
        component = shallow(<CarouselNavigation direction={"left"} />);
    });

    it('should have 1 child', () => {
        expect(component.dive().find('div').children()).toHaveLength(1);
    });

});
