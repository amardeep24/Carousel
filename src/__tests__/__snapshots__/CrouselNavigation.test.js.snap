// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1::The CarouselNavigation component should match the snapshot should match snapshot 1`] = `ReactWrapper {}`;
